### PR TITLE
Fix+test DTI/TDI/PI add/sub with ndarray[datetime64/timedelta64]

### PR DESCRIPTION
**Pull request by jbrockmendel**
Two notes for upcoming passes:

- the tests are pretty redundant, can be parametrized more tightly once they are all OK.
- catching ndarray[datetime64] separately from DatetimeIndex is kind of weird, but for the moment necesarry.  Once #19835 goes through and re-orders the catching, we'll be able to fix this.

### CODE REVIEW

**Comment by gfyoung** [comment]
@jbrockmendel : Good direction so far!  Looks you can parametrize now since all the CI is green.

File: pandas/core/indexes/datetimelike.py [hunk 1]
```diff
@@ -744,6 +752,11 @@ def __sub__(self, other):
         cls.__sub__ = __sub__
 
         def __rsub__(self, other):
+            if is_datetime64_dtype(other) and is_timedelta64_dtype(self):
+                # ndarray[datetime64] cannot be subtracted from self, so
+                # we need to wrap in DatetimeIndex and flip the operation
+                from pandas.core.indexes.datetimes import DatetimeIndex
```
Comment (jreback) [inline pandas/core/indexes/datetimelike.py]: since inside a function, can just import from pandas directrly

Comment (jreback) [reply pandas/core/indexes/datetimelike.py]: atually pls change all imports that are inside functions to do this (its just shorter)

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
needs a rebase. does this fix any open issue?

**Comment by jbrockmendel** [comment]
> does this fix any open issue?

Not that I'm aware of.

File: pandas/tests/indexes/datetimes/test_arithmetic.py [hunk 2]
```diff
@@ -571,6 +571,62 @@ def test_add_datetimelike_and_dti_tz(self, addend):
         with tm.assert_raises_regex(TypeError, msg):
             addend + dti_tz
 
+    # -------------------------------------------------------------
+    # __add__/__sub__ with ndarray[datetime64] and ndarray[timedelta64]
+
+    def test_dti_add_dt64_array_raises(self, tz):
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_arithmetic.py]: these tests don't exist already? odd

**Review by jreback — CHANGES_REQUESTED** [review]
this needs a whatsnew note to show what is a bug fix  / enhancement / api change (IOW what works now that didn't work before)

**Comment by jbrockmendel** [comment]
Here too Travis error is pyarrow import

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**